frontend/catalyst/api_extensions/error_mitigation.py

```diff
@@ -30,8 +30,15 @@
 from catalyst.jax_primitives import zne_p
 
 
+def polynomial_extrapolation(degree):
+    """utility to generate polynomial fitting functions of arbitrary degree"""
+    return lambda scale_factors, results: qml.transforms.poly_extrapolate(
```

Comment:
Is pennylane's `poly_extrapolate` written in JAX already?

Reply:
Looks like the [`poly_extrapolate`](https://github.com/PennyLaneAI/pennylane/blob/2450f169e14efb70c1caf86803d20aff42834bc1/pennylane/transforms/mitigate.py#L267) function calls [`_polyfit`](https://github.com/PennyLaneAI/pennylane/blob/2450f169e14efb70c1caf86803d20aff42834bc1/pennylane/transforms/mitigate.py#L246) which is written using `qml.math` calls. I'll let someone from the Xanadu team confirm, but the fact that the tests pass while using `qjit` indicates to me that the implementation is safe to use.